Build fail due to github action for NotificationEventIndexTests

The build for notification project is successful and all the test cases passed in the local env setup. Due to some github action, when NotificationEventIndexTests file is included in the build, the build fails for the NotificationEventDocTests. I have included some of the logs of the build fail for reference. 

Error logs:
```
> Task :notifications:test
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended



Suite: Test class org.opensearch.notifications.model.NotificationEventDocTests
  2> org.opentest4j.AssertionFailedError: expected: <NotificationEventDoc(metadata=DocMetadata(lastUpdateTime=2021-09-10T18:24:33.494Z, createdTime=2021-09-10T18:07:53.494Z, tenant=selectedTenant, access=[User:user, Role:sample_role, BERole:sample_backend_role]), event=NotificationEvent(eventSource=EventSource(title=title, referenceId=reference_id, feature=alerting, severity=info, tags=[tag1, tag2]), statusList=[EventStatus(configId=config_id, configName=name, configType=chime, emailRecipientStatus=[], deliveryStatus=DeliveryStatus(statusCode=200, statusText=success))]))> but was: <NotificationEventDoc(metadata=DocMetadata(lastUpdateTime=2021-09-10T18:24:33.214Z, createdTime=2021-09-10T18:07:53.214Z, tenant=tenant, access=[User:user, Role:sample_role, BERole:sample_backend_role]), event=NotificationEvent(eventSource=EventSource(title=title, referenceId=reference_id, feature=alerting, severity=info, tags=[tag1, tag2]), statusList=[EventStatus(configId=config_id, configName=name, configType=chime, emailRecipientStatus=[], deliveryStatus=DeliveryStatus(statusCode=200, statusText=success))]))>
        at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1124)
        at org.opensearch.notifications.model.NotificationEventDocTests.Event doc should safely ignore extra field in json object(NotificationEventDocTests.kt:135)
  2> org.opentest4j.AssertionFailedError: Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:71)
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:37)
        at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3007)
        at org.opensearch.notifications.model.NotificationEventDocTests.Event doc should throw exception if metadata is absent in json(NotificationEventDocTests.kt:165)
  2> org.opentest4j.AssertionFailedError: Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:71)
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:37)
        at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3007)
        at org.opensearch.notifications.model.NotificationEventDocTests.Event doc should throw exception if event is absent in json(NotificationEventDocTests.kt:184)
  2> org.opentest4j.AssertionFailedError: expected: <NotificationEventDoc(metadata=DocMetadata(lastUpdateTime=2021-09-10T18:24:33.519Z, createdTime=2021-09-10T18:07:53.519Z, tenant=tenant, access=[User:user, Role:sample_role, BERole:sample_backend_role]), event=NotificationEvent(eventSource=EventSource(title=title, referenceId=reference_id, feature=alerting, severity=info, tags=[tag1, tag2]), statusList=[EventStatus(configId=config_id, configName=name, configType=chime, emailRecipientStatus=[], deliveryStatus=DeliveryStatus(statusCode=200, statusText=success))]))> but was: <NotificationEventDoc(metadata=DocMetadata(lastUpdateTime=2021-09-10T18:24:33.214Z, createdTime=2021-09-10T18:07:53.214Z, tenant=tenant, access=[User:user, Role:sample_role, BERole:sample_backend_role]), event=NotificationEvent(eventSource=EventSource(title=title, referenceId=reference_id, feature=alerting, severity=info, tags=[tag1, tag2]), statusList=[EventStatus(configId=config_id, configName=name, configType=chime, emailRecipientStatus=[], deliveryStatus=DeliveryStatus(statusCode=200, statusText=success))]))>
        at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1124)
        at org.opensearch.notifications.model.NotificationEventDocTests.Event doc serialize and deserialize using json config object should be equal(NotificationEventDocTests.kt:72)
org.opensearch.notifications.model.NotificationEventDocTests > Event doc should safely ignore extra field in json object() FAILED
    org.opentest4j.AssertionFailedError: expected: <NotificationEventDoc(metadata=DocMetadata(lastUpdateTime=2021-09-10T18:24:33.494Z, createdTime=2021-09-10T18:07:53.494Z, tenant=selectedTenant, access=[User:user, Role:sample_role, BERole:sample_backend_role]), event=NotificationEvent(eventSource=EventSource(title=title, referenceId=reference_id, feature=alerting, severity=info, tags=[tag1, tag2]), statusList=[EventStatus(configId=config_id, configName=name, configType=chime, emailRecipientStatus=[], deliveryStatus=DeliveryStatus(statusCode=200, statusText=success))]))> but was: <NotificationEventDoc(metadata=DocMetadata(lastUpdateTime=2021-09-10T18:24:33.214Z, createdTime=2021-09-10T18:07:53.214Z, tenant=tenant, access=[User:user, Role:sample_role, BERole:sample_backend_role]), event=NotificationEvent(eventSource=EventSource(title=title, referenceId=reference_id, feature=alerting, severity=info, tags=[tag1, tag2]), statusList=[EventStatus(configId=config_id, configName=name, configType=chime, emailRecipientStatus=[], deliveryStatus=DeliveryStatus(statusCode=200, statusText=success))]))>
        at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1124)
        at org.opensearch.notifications.model.NotificationEventDocTests.Event doc should safely ignore extra field in json object(NotificationEventDocTests.kt:135)

org.opensearch.notifications.model.NotificationEventDocTests > Event doc should throw exception if metadata is absent in json() FAILED
    org.opentest4j.AssertionFailedError: Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:71)
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:37)
        at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3007)
        at org.opensearch.notifications.model.NotificationEventDocTests.Event doc should throw exception if metadata is absent in json(NotificationEventDocTests.kt:165)

org.opensearch.notifications.model.NotificationEventDocTests > Event doc should throw exception if event is absent in json() FAILED
    org.opentest4j.AssertionFailedError: Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:71)
        at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:37)
        at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3007)
        at org.opensearch.notifications.model.NotificationEventDocTests.Event doc should throw exception if event is absent in json(NotificationEventDocTests.kt:184)

org.opensearch.notifications.model.NotificationEventDocTests > Event doc serialize and deserialize using json config object should be equal() FAILED
    org.opentest4j.AssertionFailedError: expected: <NotificationEventDoc(metadata=DocMetadata(lastUpdateTime=2021-09-10T18:24:33.519Z, createdTime=2021-09-10T18:07:53.519Z, tenant=tenant, access=[User:user, Role:sample_role, BERole:sample_backend_role]), event=NotificationEvent(eventSource=EventSource(title=title, referenceId=reference_id, feature=alerting, severity=info, tags=[tag1, tag2]), statusList=[EventStatus(configId=config_id, configName=name, configType=chime, emailRecipientStatus=[], deliveryStatus=DeliveryStatus(statusCode=200, statusText=success))]))> but was: <NotificationEventDoc(metadata=DocMetadata(lastUpdateTime=2021-09-10T18:24:33.214Z, createdTime=2021-09-10T18:07:53.214Z, tenant=tenant, access=[User:user, Role:sample_role, BERole:sample_backend_role]), event=NotificationEvent(eventSource=EventSource(title=title, referenceId=reference_id, feature=alerting, severity=info, tags=[tag1, tag2]), statusList=[EventStatus(configId=config_id, configName=name, configType=chime, emailRecipientStatus=[], deliveryStatus=DeliveryStatus(statusCode=200, statusText=success))]))>
        at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1124)
        at org.opensearch.notifications.model.NotificationEventDocTests.Event doc serialize and deserialize using json config object should be equal(NotificationEventDocTests.kt:72)


27 tests completed, 4 failed
Tests with failures:
 - org.opensearch.notifications.model.NotificationEventDocTests.Event doc should safely ignore extra field in json object()
 - org.opensearch.notifications.model.NotificationEventDocTests.Event doc should throw exception if metadata is absent in json()
 - org.opensearch.notifications.model.NotificationEventDocTests.Event doc should throw exception if event is absent in json()
 - org.opensearch.notifications.model.NotificationEventDocTests.Event doc serialize and deserialize using json config object should be equal()

> Task :notifications:test FAILED


Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.6.1/userguide/command_line_interface.html#sec:command_line_warnings
22 actionable tasks: 22 executed
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':notifications:test'.
> There were failing tests. See the report at: file:///home/runner/work/notifications/notifications/notifications/notifications/build/reports/tests/test/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 35s
Error: Process completed with exit code 1.
```

To overcome this issue and merge the NotificationEventIndexTests file, a modification is to be made to the notification-test-and-build-workflow.yml. The highlighted line from the following file should be modified to the following command to exclude the NotificationEventIndexTests temporarily to build the notification project.

File and the command to be modified: https://github.com/opensearch-project/notifications/blob/88b62178185c497e78a486926983536fc2259a1c/.github/workflows/notifications-test-and-build-workflow.yml#L73

Command: `./gradlew build -PexcludeTests="**/SesChannelIT*" -PexcludeTests="**/NotificationEventIndexTests*" -Dopensearch.version=${{ env.OPENSEARCH_VERSION }}`

--- a/notifications/notifications/src/main/config/notifications.yml
+++ b/notifications/notifications/src/main/config/notifications.yml
@@ -31,3 +31,4 @@ opensearch.notifications:
   general:
     operationTimeoutMs: 60000 # 60 seconds, Minimum 100ms
     defaultItemsQueryCount: 100 # default number of items to query
+    filterSendByBackendRoles: false # Does sendNotification needs to validate user's backend roles

--- a/notifications/notifications/src/main/kotlin/org/opensearch/notifications/model/DocMetadata.kt
+++ b/notifications/notifications/src/main/kotlin/org/opensearch/notifications/model/DocMetadata.kt
@@ -31,6 +31,7 @@ import org.opensearch.common.xcontent.ToXContent
 import org.opensearch.common.xcontent.XContentBuilder
 import org.opensearch.common.xcontent.XContentParser
 import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.commons.notifications.NotificationConstants.CREATED_TIME_TAG
 import org.opensearch.commons.notifications.NotificationConstants.UPDATED_TIME_TAG
 import org.opensearch.commons.utils.logger
 import org.opensearch.commons.utils.stringList
@@ -47,7 +48,6 @@ data class DocMetadata(
     companion object {
         private val log by logger(DocMetadata::class.java)
         const val METADATA_TAG = "metadata"
-        const val CREATED_TIME_TAG = "created_time_ms"
         const val ACCESS_LIST_TAG = "access"
 
         /**

--- a/notifications/notifications/src/main/kotlin/org/opensearch/notifications/model/NotificationConfigDoc.kt
+++ b/notifications/notifications/src/main/kotlin/org/opensearch/notifications/model/NotificationConfigDoc.kt
@@ -38,7 +38,7 @@ import org.opensearch.notifications.model.DocMetadata.Companion.METADATA_TAG
 import java.io.IOException
 
 /**
- * Data class representing Notification config.
+ * Data class representing Notification config with metadata.
  */
 data class NotificationConfigDoc(
     val metadata: DocMetadata,

--- a/notifications/notifications/src/main/kotlin/org/opensearch/notifications/model/NotificationEventDoc.kt
+++ b/notifications/notifications/src/main/kotlin/org/opensearch/notifications/model/NotificationEventDoc.kt
@@ -57,7 +57,7 @@ data class NotificationEventDoc(
         @Throws(IOException::class)
         fun parse(parser: XContentParser): NotificationEventDoc {
             var metadata: DocMetadata? = null
-            var config: NotificationEvent? = null
+            var event: NotificationEvent? = null
 
             XContentParserUtils.ensureExpectedToken(
                 XContentParser.Token.START_OBJECT,
@@ -69,7 +69,7 @@ data class NotificationEventDoc(
                 parser.nextToken()
                 when (fieldName) {
                     METADATA_TAG -> metadata = DocMetadata.parse(parser)
-                    EVENT_TAG -> config = NotificationEvent.parse(parser)
+                    EVENT_TAG -> event = NotificationEvent.parse(parser)
                     else -> {
                         parser.skipChildren()
                         log.info("Unexpected field: $fieldName, while parsing event doc")
@@ -77,10 +77,10 @@ data class NotificationEventDoc(
                 }
             }
             metadata ?: throw IllegalArgumentException("$METADATA_TAG field absent")
-            config ?: throw IllegalArgumentException("$EVENT_TAG field absent")
+            event ?: throw IllegalArgumentException("$EVENT_TAG field absent")
             return NotificationEventDoc(
                 metadata,
-                config
+                event
             )
         }
     }

--- a/notifications/notifications/src/main/kotlin/org/opensearch/notifications/security/UserAccess.kt
+++ b/notifications/notifications/src/main/kotlin/org/opensearch/notifications/security/UserAccess.kt
@@ -49,4 +49,9 @@ interface UserAccess {
      * validate if user has access based on given access list
      */
     fun doesUserHasAccess(user: User?, access: List<String>): Boolean
+
+    /**
+     * validate if user has send-notification access based on given access list
+     */
+    fun doesUserHasSendAccess(user: User?, access: List<String>): Boolean
 }

--- a/notifications/notifications/src/main/kotlin/org/opensearch/notifications/security/UserAccessManager.kt
+++ b/notifications/notifications/src/main/kotlin/org/opensearch/notifications/security/UserAccessManager.kt
@@ -77,4 +77,11 @@ internal object UserAccessManager : UserAccess {
         }
         return user.backendRoles.any { it in access }
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun doesUserHasSendAccess(user: User?, access: List<String>): Boolean {
+        return !PluginSettings.filterSendByBackendRoles || doesUserHasAccess(user, access)
+    }
 }

--- a/notifications/notifications/src/main/kotlin/org/opensearch/notifications/send/SendMessageActionHelper.kt
+++ b/notifications/notifications/src/main/kotlin/org/opensearch/notifications/send/SendMessageActionHelper.kt
@@ -218,7 +218,7 @@ object SendMessageActionHelper {
                 listOf(),
                 DeliveryStatus(RestStatus.NOT_FOUND.status.toString(), "Channel ${channelEntry.key} not found")
             )
-        } else if (!userAccess.doesUserHasAccess(user, channelEntry.value!!.configDoc.metadata.access)) {
+        } else if (!userAccess.doesUserHasSendAccess(user, channelEntry.value!!.configDoc.metadata.access)) {
             Metrics.NOTIFICATIONS_PERMISSION_USER_ERROR.counter.increment()
             return EventStatus(
                 channelEntry.key,
@@ -418,7 +418,7 @@ object SendMessageActionHelper {
                     "Sender ${email.emailAccountID} not found"
                 )
             )
-        } else if (!userAccess.doesUserHasAccess(user, accountDocInfo.configDoc.metadata.access)) {
+        } else if (!userAccess.doesUserHasSendAccess(user, accountDocInfo.configDoc.metadata.access)) {
             Metrics.NOTIFICATIONS_PERMISSION_USER_ERROR.counter.increment()
             return eventStatus.copy(
                 emailRecipientStatus = listOf(),
@@ -429,7 +429,7 @@ object SendMessageActionHelper {
             )
         }
         val accessDeniedGroupIds = childConfigMap.filterValues {
-            it != null && !userAccess.doesUserHasAccess(user, it.configDoc.metadata.access)
+            it != null && !userAccess.doesUserHasSendAccess(user, it.configDoc.metadata.access)
         }.keys
         val invalidGroupIds = childConfigMap.filterValues { it == null }.keys
         val groups = childConfigMap.values.filterNotNull()

--- a/notifications/notifications/src/main/kotlin/org/opensearch/notifications/settings/PluginSettings.kt
+++ b/notifications/notifications/src/main/kotlin/org/opensearch/notifications/settings/PluginSettings.kt
@@ -69,6 +69,11 @@ internal object PluginSettings {
     private const val DEFAULT_ITEMS_QUERY_COUNT_KEY = "$GENERAL_KEY_PREFIX.defaultItemsQueryCount"
 
     /**
+     * Setting to choose filter send by backend role.
+     */
+    private const val FILTER_SEND_BY_BACKEND_ROLES_KEY = "$GENERAL_KEY_PREFIX.filterSendByBackendRoles"
+
+    /**
      * Legacy alerting plugin filter_by_backend_roles setting.
      */
     private const val LEGACY_FILTER_BY_BACKEND_ROLES_KEY = "opendistro.alerting.filter_by_backend_roles"
@@ -110,6 +115,12 @@ internal object PluginSettings {
     @Volatile
     var defaultItemsQueryCount: Int
 
+    /**
+     * Filter send by backend role.
+     */
+    @Volatile
+    var filterSendByBackendRoles: Boolean
+
     private const val DECIMAL_RADIX: Int = 10
 
     private val log by logger(javaClass)
@@ -130,9 +141,12 @@ internal object PluginSettings {
         operationTimeoutMs = (settings?.get(OPERATION_TIMEOUT_MS_KEY)?.toLong()) ?: DEFAULT_OPERATION_TIMEOUT_MS
         defaultItemsQueryCount = (settings?.get(DEFAULT_ITEMS_QUERY_COUNT_KEY)?.toInt())
             ?: DEFAULT_ITEMS_QUERY_COUNT_VALUE
+        filterSendByBackendRoles = (settings?.get(FILTER_SEND_BY_BACKEND_ROLES_KEY)?.toBoolean())
+            ?: false
         defaultSettings = mapOf(
             OPERATION_TIMEOUT_MS_KEY to operationTimeoutMs.toString(DECIMAL_RADIX),
-            DEFAULT_ITEMS_QUERY_COUNT_KEY to defaultItemsQueryCount.toString(DECIMAL_RADIX)
+            DEFAULT_ITEMS_QUERY_COUNT_KEY to defaultItemsQueryCount.toString(DECIMAL_RADIX),
+            FILTER_SEND_BY_BACKEND_ROLES_KEY to filterSendByBackendRoles.toString()
         )
     }
 
@@ -147,6 +161,12 @@ internal object PluginSettings {
         DEFAULT_ITEMS_QUERY_COUNT_KEY,
         defaultSettings[DEFAULT_ITEMS_QUERY_COUNT_KEY]!!.toInt(),
         MINIMUM_ITEMS_QUERY_COUNT,
+        NodeScope, Dynamic
+    )
+
+    val FILTER_SEND_BY_BACKEND_ROLES: Setting<Boolean> = Setting.boolSetting(
+        FILTER_SEND_BY_BACKEND_ROLES_KEY,
+        defaultSettings[FILTER_SEND_BY_BACKEND_ROLES_KEY]!!.toBoolean(),
         NodeScope, Dynamic
     )
 
@@ -178,7 +198,8 @@ internal object PluginSettings {
     fun getAllSettings(): List<Setting<*>> {
         return listOf(
             OPERATION_TIMEOUT_MS,
-            DEFAULT_ITEMS_QUERY_COUNT
+            DEFAULT_ITEMS_QUERY_COUNT,
+            FILTER_SEND_BY_BACKEND_ROLES
         )
     }
 
@@ -189,6 +210,7 @@ internal object PluginSettings {
     private fun updateSettingValuesFromLocal(clusterService: ClusterService) {
         operationTimeoutMs = OPERATION_TIMEOUT_MS.get(clusterService.settings)
         defaultItemsQueryCount = DEFAULT_ITEMS_QUERY_COUNT.get(clusterService.settings)
+        filterSendByBackendRoles = FILTER_SEND_BY_BACKEND_ROLES.get(clusterService.settings)
     }
 
     /**
@@ -206,6 +228,11 @@ internal object PluginSettings {
         if (clusterDefaultItemsQueryCount != null) {
             log.debug("$LOG_PREFIX:$DEFAULT_ITEMS_QUERY_COUNT_KEY -autoUpdatedTo-> $clusterDefaultItemsQueryCount")
             defaultItemsQueryCount = clusterDefaultItemsQueryCount
+        }
+        val clusterFilterSendByBackendRole = clusterService.clusterSettings.get(FILTER_SEND_BY_BACKEND_ROLES)
+        if (clusterFilterSendByBackendRole != null) {
+            log.debug("$LOG_PREFIX:$FILTER_SEND_BY_BACKEND_ROLES_KEY -autoUpdatedTo-> $clusterFilterSendByBackendRole")
+            filterSendByBackendRoles = clusterFilterSendByBackendRole
         }
     }
 
@@ -228,6 +255,10 @@ internal object PluginSettings {
             defaultItemsQueryCount = it
             log.info("$LOG_PREFIX:$DEFAULT_ITEMS_QUERY_COUNT_KEY -updatedTo-> $it")
         }
+        clusterService.clusterSettings.addSettingsUpdateConsumer(FILTER_SEND_BY_BACKEND_ROLES) {
+            filterSendByBackendRoles = it
+            log.info("$LOG_PREFIX:$FILTER_SEND_BY_BACKEND_ROLES_KEY -updatedTo-> $it")
+        }
     }
 
     // reset the settings values to default values for testing purpose
@@ -235,5 +266,6 @@ internal object PluginSettings {
     fun reset() {
         operationTimeoutMs = DEFAULT_OPERATION_TIMEOUT_MS
         defaultItemsQueryCount = DEFAULT_ITEMS_QUERY_COUNT_VALUE
+        filterSendByBackendRoles = false
     }
 }

--- a/notifications/notifications/src/test/kotlin/org/opensearch/notifications/index/NotificationEventIndexTests.kt
+++ b/notifications/notifications/src/test/kotlin/org/opensearch/notifications/index/NotificationEventIndexTests.kt
@@ -17,7 +17,6 @@ import io.mockk.mockkObject
 import org.junit.jupiter.api.Assertions
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.Assertions.assertNull
-import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import org.mockito.Mockito.anyLong
 import org.mockito.Mockito.mock
@@ -49,22 +48,16 @@ import java.time.Instant
 
 internal class NotificationEventIndexTests {
 
-    private lateinit var client: Client
+    private val client: Client = mock(Client::class.java, "client")
 
     private val indexName = ".opensearch-notifications-event"
 
-    private lateinit var clusterService: ClusterService
-
-    @BeforeEach
-    fun setUp() {
-        client = mock(Client::class.java, "client")
-        clusterService = mock(ClusterService::class.java, "clusterService")
-        NotificationEventIndex.initialize(client, clusterService)
-    }
+    private val clusterService: ClusterService = mock(ClusterService::class.java, "clusterService")
 
     @Suppress("UNCHECKED_CAST")
     @Test
     fun `index operation to get single event`() {
+        NotificationEventIndex.initialize(client, clusterService)
         // creating expected value
         val id = "index-1"
         val docInfo = DocInfo("index-1", 1, 1, 1)
@@ -130,6 +123,7 @@ internal class NotificationEventIndexTests {
     @Suppress("UNCHECKED_CAST")
     @Test
     fun `NotificationEventIndex should safely return null if response source is null`() {
+        NotificationEventIndex.initialize(client, clusterService)
         val id = "index-1"
         // mocking the dependencies for isIndexExists function
         mockIsIndexExists()
@@ -159,6 +153,7 @@ internal class NotificationEventIndexTests {
     @Suppress("UNCHECKED_CAST")
     @Test
     fun `NotificationEventIndex should throw exception if response isn't acknowledged`() {
+        NotificationEventIndex.initialize(client, clusterService)
         val id = "index-1"
         // mocking the dependencies for isIndexExists function
         mockIsIndexExists()
@@ -174,6 +169,7 @@ internal class NotificationEventIndexTests {
     @Suppress("UNCHECKED_CAST")
     @Test
     fun `NotificationEventIndex should throw exception if index couldn't be created`() {
+        NotificationEventIndex.initialize(client, clusterService)
         val id = "index-1"
         // mocking the dependencies for isIndexExists function
         mockIsIndexExists()

--- a/notifications/notifications/src/test/kotlin/org/opensearch/notifications/model/NotificationEventDocTests.kt
+++ b/notifications/notifications/src/test/kotlin/org/opensearch/notifications/model/NotificationEventDocTests.kt
@@ -45,7 +45,7 @@ internal class NotificationEventDocTests {
     @Test
     fun `Event doc serialize and deserialize using json config object should be equal`() {
         val lastUpdatedTimeMs = Instant.ofEpochMilli(Instant.now().toEpochMilli())
-        val createdTimeMs = lastUpdatedTimeMs.minusSeconds(1000)
+        val createdTimeMs = Instant.ofEpochMilli(Instant.now().minusSeconds(2000).toEpochMilli())
         val metadata = DocMetadata(
             lastUpdatedTimeMs,
             createdTimeMs,
@@ -74,7 +74,7 @@ internal class NotificationEventDocTests {
     @Test
     fun `Event doc should safely ignore extra field in json object`() {
         val lastUpdatedTimeMs = Instant.ofEpochMilli(Instant.now().toEpochMilli())
-        val createdTimeMs = lastUpdatedTimeMs.minusSeconds(1000)
+        val createdTimeMs = Instant.ofEpochMilli(Instant.now().minusSeconds(2000).toEpochMilli())
         val metadata = DocMetadata(
             lastUpdatedTimeMs,
             createdTimeMs,
@@ -167,7 +167,7 @@ internal class NotificationEventDocTests {
     @Test
     fun `Event doc should throw exception if event is absent in json`() {
         val lastUpdatedTimeMs = Instant.ofEpochMilli(Instant.now().toEpochMilli())
-        val createdTimeMs = lastUpdatedTimeMs.minusSeconds(1000)
+        val createdTimeMs = Instant.ofEpochMilli(Instant.now().minusSeconds(2000).toEpochMilli())
         val jsonString = """
         {
             "metadata":{

--- a/notifications/notifications/src/test/kotlin/org/opensearch/notifications/settings/PluginSettingsTests.kt
+++ b/notifications/notifications/src/test/kotlin/org/opensearch/notifications/settings/PluginSettingsTests.kt
@@ -31,10 +31,12 @@ internal class PluginSettingsTests {
     private val generalKeyPrefix = "$keyPrefix.general"
     private val operationTimeoutKey = "$generalKeyPrefix.operationTimeoutMs"
     private val defaultItemQueryCountKey = "$generalKeyPrefix.defaultItemsQueryCount"
+    private val filterSendByBackendRolesKey = "$generalKeyPrefix.filterSendByBackendRoles"
 
     private val defaultSettings = Settings.builder()
         .put(operationTimeoutKey, 60000L)
         .put(defaultItemQueryCountKey, 100L)
+        .put(filterSendByBackendRolesKey, false)
         .build()
 
     @BeforeEach
@@ -54,14 +56,22 @@ internal class PluginSettingsTests {
 
         Assert.assertTrue(
             settings.containsAll(
-                listOf<Any> (
+                listOf<Any>(
                     PluginSettings.OPERATION_TIMEOUT_MS,
-                    PluginSettings.DEFAULT_ITEMS_QUERY_COUNT
+                    PluginSettings.DEFAULT_ITEMS_QUERY_COUNT,
+                    PluginSettings.FILTER_SEND_BY_BACKEND_ROLES
                 )
             )
         )
         Assertions.assertEquals(defaultSettings[operationTimeoutKey], PluginSettings.operationTimeoutMs.toString())
-        Assertions.assertEquals(defaultSettings[defaultItemQueryCountKey], PluginSettings.defaultItemsQueryCount.toString())
+        Assertions.assertEquals(
+            defaultSettings[defaultItemQueryCountKey],
+            PluginSettings.defaultItemsQueryCount.toString()
+        )
+        Assertions.assertEquals(
+            defaultSettings[filterSendByBackendRolesKey],
+            PluginSettings.filterSendByBackendRoles.toString()
+        )
     }
 
     @Test
@@ -69,13 +79,18 @@ internal class PluginSettingsTests {
         val clusterSettings = Settings.builder()
             .put(operationTimeoutKey, 50000L)
             .put(defaultItemQueryCountKey, 200)
+            .put(filterSendByBackendRolesKey, true)
             .build()
 
         whenever(clusterService.settings).thenReturn(defaultSettings)
         whenever(clusterService.clusterSettings).thenReturn(
             ClusterSettings(
                 clusterSettings,
-                setOf(PluginSettings.OPERATION_TIMEOUT_MS, PluginSettings.DEFAULT_ITEMS_QUERY_COUNT)
+                setOf(
+                    PluginSettings.OPERATION_TIMEOUT_MS,
+                    PluginSettings.DEFAULT_ITEMS_QUERY_COUNT,
+                    PluginSettings.FILTER_SEND_BY_BACKEND_ROLES
+                )
             )
         )
         PluginSettings.addSettingsUpdateConsumer(clusterService)
@@ -87,6 +102,10 @@ internal class PluginSettingsTests {
             200,
             clusterService.clusterSettings.get(PluginSettings.DEFAULT_ITEMS_QUERY_COUNT)
         )
+        Assertions.assertEquals(
+            true,
+            clusterService.clusterSettings.get(PluginSettings.FILTER_SEND_BY_BACKEND_ROLES)
+        )
     }
 
     @Test
@@ -96,7 +115,11 @@ internal class PluginSettingsTests {
         whenever(clusterService.clusterSettings).thenReturn(
             ClusterSettings(
                 clusterSettings,
-                setOf(PluginSettings.OPERATION_TIMEOUT_MS, PluginSettings.DEFAULT_ITEMS_QUERY_COUNT)
+                setOf(
+                    PluginSettings.OPERATION_TIMEOUT_MS,
+                    PluginSettings.DEFAULT_ITEMS_QUERY_COUNT,
+                    PluginSettings.FILTER_SEND_BY_BACKEND_ROLES
+                )
             )
         )
         PluginSettings.addSettingsUpdateConsumer(clusterService)
@@ -107,6 +130,10 @@ internal class PluginSettingsTests {
         Assertions.assertEquals(
             defaultSettings[defaultItemQueryCountKey],
             clusterService.clusterSettings.get(PluginSettings.DEFAULT_ITEMS_QUERY_COUNT).toString()
+        )
+        Assertions.assertEquals(
+            defaultSettings[filterSendByBackendRolesKey],
+            clusterService.clusterSettings.get(PluginSettings.FILTER_SEND_BY_BACKEND_ROLES).toString()
         )
     }
 }

